### PR TITLE
Improve performance of GetFacilityResults

### LINKED ISSUE #2646: Improve `getFacilityResults` Performance
This GraphQL query results in some very long-running database queries. We need to fix it.

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- Resolves #2646 

## Changes Proposed

- Add an index on `patient_link.test_order_id`
- Set the TestOrder relationship on `patient_link` to be lazily loaded 

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -3052,3 +3052,14 @@ databaseChangeLog:
               - column: *created_at_column
         - sql: |
             GRANT SELECT ON TABLE ${database.defaultSchemaName}.report_stream_response TO ${noPhiUsername};
+  - changeSet:
```
Comment (zdeveloper) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:3055]: add db tag

**Review by zdeveloper — COMMENTED** [review]
Missing DB tag for liquibase migration.

**Review by bmaden-usds — APPROVED** [review]
LGTM, and looking forward to seeing some shrinking response times!

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (nullflux) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:3055]: done!

**Pull request merged**